src/pool/cli.c

```diff
@@ -1910,13 +1910,14 @@ map_refresh(tse_task_t *task)
 			DP_UUID(pool->dp_pool), task, pool->dp_map_task);
 		arg->mra_passive = true;
 		rc = tse_task_register_deps(task, 1, &pool->dp_map_task);
-		D_RWLOCK_UNLOCK(&pool->dp_map_lock);
 		if (rc != 0) {
 			D_ERROR(DF_UUID": failed to depend on active pool map "
 				"refresh task: "DF_RC"\n", DP_UUID(pool->dp_pool), DP_RC(rc));
+			D_RWLOCK_UNLOCK(&pool->dp_map_lock);
```

Comment:
Suggest to move the UNLOCK to ahead of D_ERROR() that may be blocked inside d_log().

Reply:
OK, I can change it in following patch, thx